DeviceBridge/Services/SubscriptionCallbackFactory.cs

```diff
@@ -0,0 +1,203 @@
+                    using var httpResponse = await _httpClientFactory.CreateClient("RetryClient").PostAsync(messageSubscription.CallbackUrl, requestPayload);
+                    var statusCode = (int)httpResponse.StatusCode;
+                    if (statusCode >= 200 && statusCode < 300)
+                    {
+                        _logger.Info("Received C2D message callback with status {statusCode}, request accepted.", statusCode);
+                        return ReceiveMessageCallbackStatus.Accept;
+                    }
+
+                    if (statusCode >= 400 && statusCode < 500)
+                    {
+                        _logger.Info("Received C2D message callback with status {statusCode}, request rejected.", statusCode);
+                        return ReceiveMessageCallbackStatus.Reject;
+                    }
+
+                    _logger.Info("Received C2D message callback with status {statusCode}, request abandoned.", statusCode);
+                    return ReceiveMessageCallbackStatus.Abandon;
+                }
+                catch (Exception e)
+                {
+                    _logger.Error(e, "Failed to execute message callback, device {deviceId}. Request abandoned.", deviceId);
+                    return ReceiveMessageCallbackStatus.Abandon;
+                }
+            };
+        }
+
+        public MethodCallback GetMethodCallback(string deviceId, DeviceSubscription methodSubscription)
+        {
+            return async (methodRequest, _) =>
+            {
+                _logger.Info("Got method request for device {deviceId}. Callback URL {callbackUrl}. Method: {methodName}. Payload: {payload}", deviceId, methodSubscription.CallbackUrl, methodRequest.Name, methodRequest.DataAsJson);
+
+                try
+                {
+                    var body = new MethodInvocationEventBody()
+                    {
+                        DeviceId = deviceId,
+                        DeviceReceivedAt = DateTime.UtcNow,
+                        MethodName = methodRequest.Name,
+                        RequestData = new JRaw(methodRequest.DataAsJson),
+                    };
+
+                    // Send request to callback URL
+                    var requestPayload = new StringContent(JsonConvert.SerializeObject(body), Encoding.UTF8, "application/json");
+                    using var httpResponse = await _httpClientFactory.CreateClient("RetryClient").PostAsync(methodSubscription.CallbackUrl, requestPayload);
+                    httpResponse.EnsureSuccessStatusCode();
+
+                    // Read method response from callback response
+                    using var responseStream = await httpResponse.Content.ReadAsStreamAsync();
+                    MethodResponseBody responseBody = null;
+
+                    try
+                    {
+                        responseBody = await System.Text.Json.JsonSerializer.DeserializeAsync<MethodResponseBody>(responseStream, new JsonSerializerOptions
```

Comment:
Add System.Text.Json to using

Reply:
It would conflict with `Newtonsoft.Json`. We're using `Newtonsoft.Json` for features that `System.Text.Json` still doesn't have, as recommended by MSDN.